### PR TITLE
Feature: add infrastructure for integration tests of cart api

### PR DESCRIPTION
**Pull request by YanBrovkin**
### Problem
During transition program logic with cart api interaction from Storefront to experience api module we have to control regression of functionality inside Storefront.

### Solution
It's necessary to create infrastructure for integration tests of cart api

### Proposed of changes
There are some infrastructure workaround & example test of reading empty cart

### Make sure these boxes are checked:
- [x] Check all the changes in github PR - files count (non of them are redundant, have meaningful changes, all are added), if target branch is correct
- [x] Check methods and variable namings - it should be self descriptive, no typos
- [x] Check you did not introduce breaking changes in API and public models/services.
- [x] Respect extensibility - https://community.virtocommerce.com/t/extensibility-basics-the-domain-model-and-persistence-layer-extension/141
- [x] Follow [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and [SOLID](https://en.wikipedia.org/wiki/SOLID) principles
- [x] For unit tests - follow Microsoft best practices: https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices
- [x] Consolidate solution dependencies in case you are using newer version, update VC module dependencies in module.manifest. Do not upgrade 3rd party packages that are shipped with the platform with the version newer than in the platform.
- [x] Check code style conventions - https://github.com/VirtoCommerce/styleguide/blob/master/csharp.md
- [x] Check PR have a concise and descriptive title, follow [git commit message rules](https://github.com/VirtoCommerce/styleguide/blob/master/gitcommits.md)

### CODE REVIEW

File: VirtoCommerce.Storefront.IntegrationTests/Cart/ReadingTests.cs [hunk 1]
```diff
@@ -29,6 +71,29 @@ public void GetCartItemsCount_IfCartHasItems_ShouldReturnExactItemsCount()
         {
         }
 
+        private ShoppingCart GetNewCart()
```
Comment (Vectorfield4) [inline VirtoCommerce.Storefront.IntegrationTests/Cart/ReadingTests.cs:74]: Need to move this to "object builders" or register class with fixture in ctor of base class.

File: VirtoCommerce.Storefront.IntegrationTests/Infrastructure/MoneyJsonConverter.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        public override bool CanRead => true;
+
+        public override Money ReadJson(JsonReader reader, Type objectType, Money existingValue, bool hasExistingValue, JsonSerializer serializer)
+        {
+            var obj = JObject.Load(reader);
+
+            var currencyCode = obj["currency"]["code"].Value<string>();
+            var amount = obj["amount"].Value<decimal>();
+
+            // TODO: here we should try to find incoming currency code inside available currencies,
```
Comment (Vectorfield4) [inline VirtoCommerce.Storefront.IntegrationTests/Infrastructure/MoneyJsonConverter.cs:21]: What you think about removing this comments?

File: VirtoCommerce.Storefront.IntegrationTests/Cart/ReadingTests.cs [hunk 3]
```diff
@@ -1,17 +1,59 @@
...
+                .Excluding(c => c.SelectedMemberPath == "Currency.CultureName")
+                .Excluding(c => c.SelectedMemberPath == "Currency.NumberFormat")
+                .Excluding(c => c.SelectedMemberPath == "Currency.EnglishName")
+                .Excluding(c => c.SelectedMemberInfo.Name == "CustomerId")
+                .Excluding(c => c.SelectedMemberInfo.Name == "ExtendedPriceTotal")
+                .Excluding(c => c.SelectedMemberInfo.Name == "ExtendedPriceTotalWithTax")
+                .ComparingByValue<ShoppingCart>()
+                .ComparingByMembers<User>()
+                .ComparingByMembers<Currency>()
+            );
```
[the remark this answers is not in the document]
Comment (YanBrovkin) [reply VirtoCommerce.Storefront.IntegrationTests/Cart/ReadingTests.cs]: removed

File: VirtoCommerce.Storefront.IntegrationTests/Cart/ReadingTests.cs (hunk 1 above)
Comment (YanBrovkin) [reply VirtoCommerce.Storefront.IntegrationTests/Cart/ReadingTests.cs]: removed, unnecessary

File: VirtoCommerce.Storefront.IntegrationTests/Infrastructure/MoneyJsonConverter.cs (hunk 2 above)
Comment (YanBrovkin) [reply VirtoCommerce.Storefront.IntegrationTests/Infrastructure/MoneyJsonConverter.cs]: removed

**Pull request merged**